samples/shared/SynthSound.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef SHARED_SYNTH_SOUND_H
+#define SHARED_SYNTH_SOUND_H
+
+
+#include <cstdint>
+#include <atomic>
+#include <math.h>
+#include <memory>
+#include "IRenderableAudio.h"
+
+constexpr double kDefaultFrequency = 440.0;
+constexpr int32_t kDefaultSampleRate = 48000;
+constexpr double kPi = M_PI;
+constexpr double kTwoPi = kPi * 2;
+constexpr int32_t kNumSineWaves = 5;
+constexpr int32_t kNumExtraReleasePeriods = 5;
+constexpr int32_t kAttackPeriods = 10;
+constexpr int32_t kDecayPeriods = 8;
+constexpr float kMaxAttackMultiplier = 5.0f;
+
+class SynthSound : public IRenderableAudio {
+
+public:
+
+    SynthSound() {
+        for (int i = 0; i < kNumSineWaves; i++) {
+            mFrequencies[i] = kDefaultFrequency * (i + 1);
+        }
+    }
+
+    ~SynthSound() = default;
+
+    void setWaveOn(bool isWaveOn) {
+        if (!mIsWaveOn && isWaveOn) {
+            mAttackDecayCount.store(0);
+        }
+        mIsWaveOn.store(isWaveOn);
+    };
+
+    void setSampleRate(int32_t sampleRate) {
+        mSampleRate = sampleRate;
+        updatePhaseIncrement();
+    };
+
+    void setFrequency(double frequency) {
+        for (int i = 0; i < kNumSineWaves; i++) {
+            mFrequencies[i] = frequency * (i + 1);
+        }
+        updatePhaseIncrement();
+    };
+
+    // Amplitudes from https://epubs.siam.org/doi/pdf/10.1137/S00361445003822
+    inline void setAmplitude(float amplitude) {
+        mAmplitudes[0] = amplitude * .2f;
+        mAmplitudes[1] = amplitude;
+        mAmplitudes[2] = amplitude * .1f;
```

Comment:
@philburk I followed your advice and added multiple overtones. I also added attack and delay and sustain. Can you let me know if this logic makes sense with you?

Reply:
Seems like adding a large attack and the following delay makes the sound worse. Reversed the 6th commit "https://github.com/google/oboe/pull/1354/commits/a6224c9b8374a1f6b4018fd0c35986796faf2e30"

Reply:
After doing more research, the strong attack followed by the delay makes it like a marimba instead of a piano so reverting makes sense if we wanted a synth sound instead of sounding like a marimba